catalog/core/catalog-core-attributeregistry/src/main/java/ddf/catalog/data/impl/AttributeRegistryImpl.java

```diff
@@ -43,27 +44,77 @@ public boolean register(final AttributeDescriptor attributeDescriptor) {
 
         final String name = attributeDescriptor.getName();
 
-        if (attributeMap.putIfAbsent(name, attributeDescriptor) == null) {
+        if (attributeMap.containsKey(name)) {
```

Comment:
I think there could be a race condition by checking for the existence of the attribute in the registry (`containsKey()`) and then looking it up later (`get()`). One option is to use `compute()` to perform this operation atomically:
```
MutableObject<Boolean> holder = new MutableObject<>();
attributeMap.compute(name, (attributeName, attributeValueType) -> {
    if (attributeValueType == null) { // There is no mapping
        holder.setValue(true);
        return new AttributeValueType(attributeDescriptor);
    } else {
        AttributeDescriptor comparableDescriptor = attributeValueType.getAttributeDescriptor();
        if (comparableDescriptor.equals(attributeDescriptor)) {
            // logic
        } else {
            // logic
        }
        holder.setValue(false);
        return attributeValueType;
    }
});
return holder.getValue();